includes/polling/os/sgos.inc.php

```diff
@@ -2,17 +2,105 @@
 
 use LibreNMS\RRD\RrdDefinition;
 
-$version = trim(snmp_get($device, "BLUECOAT-SG-PROXY-MIB::sgProxyVersion.0", "-OQv"), '"');
-$hardware = trim(snmp_get($device, "BLUECOAT-SG-PROXY-MIB::sgProxySoftware.0", "-OQv"), '"');
-$hostname = trim(snmp_get($device, "SNMPv2-MIB::sysName.0", "-OQv"), '"');
-$sgos_requests = snmp_get($device, "BLUECOAT-SG-PROXY-MIB::sgProxyHttpClientRequestRate.0", "-OQvU");
+$oid_list = 'sgProxyVersion.0 sgProxySoftware.0 sgProxyHttpClientRequestRate.0 sgProxyHttpClientConnections.0 sgProxyHttpClientConnectionsActive.0 sgProxyHttpClientConnectionsIdle.0 sgProxyHttpServerConnections.0 sgProxyHttpServerConnectionsActive.0 sgProxyHttpServerConnectionsIdle.0';
+$sgos = snmp_get_multi($device, $oid_list, '-OUQs', 'BLUECOAT-SG-PROXY-MIB');
+$version = $sgos[0]['sgProxyVersion.0'];
+$hardware = $sgos[0]['sgProxySoftware.0'];
 
-if (is_numeric($sgos_requests)) {
-    $rrd_def = RrdDefinition::make()->addDataset('requests', 'GAUGE', 0);
+if (is_numeric($sgos[0]['sgProxyHttpClientRequestRate'])) {
+    $tags = array(
+        'rrd_def' => RrdDefinition::make()->addDataset('requests', 'GAUGE', 0),
+    );
     $fields = array(
-        'requests' => $sgos_requests
+        'requests' => $sgos[0]['sgProxyHttpClientRequestRate'],
     );
-    $tags = compact('rrd_def');
+
     data_update($device, 'sgos_average_requests', $tags, $fields);
+
     $graphs['sgos_average_requests'] = true;
+    echo ' HTTP Req Rate';
+}
+
+if (is_numeric($sgos[0]['sgProxyHttpClientConnections'])) {
```

Comment:
If these are all new defs then you'll have to create a new file for them, you can't add additional defs without dumping and re-importing the rrd file.

Reply:
HI @laf I'm not sure I understand you on this. The new OID objects in this file create new graphs under the network section of the device. The inc files I added under html/includes/graphs/device/ deal with getting the data and displaying those RRD's. It doesn't affect the existing RRD for HTTP requests.

Reply:
Are these new OIDs just being added to the existing rrd file? If so, you can't do that and will need to create a new file with them in.

Reply:
@laf No, the new OIDs are sent to different RRD files. Like so - 

sgos_client_connections_active.rrd
sgos_client_connections_idle.rrd
sgos_client_connections.rrd
sgos_server_connections_active.rrd
sgos_server_connections_idle.rrd
sgos_server_connections.rrd

In a live deployment, they generate output like so under the 'Network' section of the ProxySG - 

![screen shot 2018-05-08 at 21 58 45](https://user-images.githubusercontent.com/25568510/39782766-17d58cac-530b-11e8-895c-85f4cb4aeb2e.png)

Reply:
Ok thanks for clarifying.